pkg/component/runtime/service.go

```diff
@@ -284,18 +284,22 @@ func (s *serviceRuntime) Run(ctx context.Context, comm Communicator) (err error)
 // .eaci.sock == Elastic Agent Connection Info socket
 const elasticAgentConnInfoSocket = ".eaci.sock"
 
-func getConnInfoServerAddress(os string, isLocal bool, port int) string {
+func getConnInfoServerAddress(os string, isLocal bool, port int, socket string) string {
 	if isLocal {
+		if socket == "" {
```

Comment:
This should error, and not start. The field should be required for a service. We don't have a default when `CPort` is missing.

Reply:
Updated